src/Collection/Collection.router.ts

```diff
@@ -233,18 +233,22 @@ export class CollectionRouter extends Router {
       )
     }
 
-    const [allCollectionsWithCount, remoteCollections] = await Promise.all([
-      this.service.getCollections({
-        q: q as string,
-        assignee: assignee as string,
-        status: status as CurationStatusFilter,
-        sort: sort as CurationStatusSort,
-        isPublished: is_published === 'true',
-        offset: page && limit ? getOffset(page, limit) : undefined,
-        limit,
-      }),
-      collectionAPI.fetchCollections(),
-    ])
+    const remoteCollections = await collectionAPI.fetchCollections(
+      toRemoteWhereCondition({ status: status as CurationStatusFilter })
+    )
+
+    const allCollectionsWithCount = await this.service.getCollections({
+      q: q as string,
+      assignee: assignee as string,
+      status: status as CurationStatusFilter,
+      sort: sort as CurationStatusSort,
+      isPublished: is_published ? is_published === 'true' : undefined,
```

Comment:
yes!
if the flag is passed, either as `true` or `false`, the query should filter those.
in case the flag is not passed, it shouldn't filter them, it should return both combinations